ports/libsimpleservo/capi/src/lib.rs

```diff
@@ -66,11 +67,19 @@ pub struct CInitOptions {
 /// The returned string is not freed. This will leak.
 #[no_mangle]
 pub extern "C" fn servo_version() -> *const c_char {
-    let v = simpleservo::servo_version();
-    let text = CString::new(v).expect("Can't create string");
-    let ptr = text.as_ptr();
-    mem::forget(text);
-    ptr
+    let mut _ptr: *const c_char = std::ptr::null();
+
+    let _result = panic::catch_unwind(AssertUnwindSafe(|| {
+        let v = simpleservo::servo_version();
+        let text = CString::new(v).expect("Can't create string");
+        _ptr = text.as_ptr();
```

Comment:
Rather than using a mutable variable, we can do the following:
```rust
let result = panic::catch_unwind(AssertUnwindSafe(|| {
    let v = simpleservo::servo_version();
    let text = CString::new(v).expect("Can't create string");
    let ptr = text.as_ptr();
    mem::forget(text)
    ptr
}));

match result {
    Ok(ptr) => ptr,
    Err(_) => std::ptr::null(),
}
```
The return value of catch_unwind contains the return value of the closure that is invoked by catch_unwind, so we don't need to do any special communication with the environment outside of the closure to extract it.

Reply:
Thank you.  Good suggestion and better understanding.